jlm/llvm/opt/alias-analyses/PointsToGraph.cpp

```diff
@@ -305,6 +440,12 @@ PointsToGraph::Node::Sources() const
   return { SourceConstIterator(Sources_.begin()), SourceConstIterator(Sources_.end()) };
 }
 
+bool
+PointsToGraph::Node::HasSource(const PointsToGraph::Node & source) const
+{
+  return Sources_.find(const_cast<Node *>(&source)) != Sources_.end();
```

Comment:
Is this const_cast necessary to get it compiling?

Reply:
Yes, sadly. The alternative is to set up a "transparent comparator", which would modify the type of `Sources_` with extra template stuff. See this [Stack Overflow Answer](https://stackoverflow.com/questions/61192917/c-stdunordered-set-find-count-contains-const-key-reference)